source/kraken/data_manager.h

```diff
@@ -96,6 +96,13 @@ class DataManager{
         ++ data_identifier;
         auto data = create_data(data_identifier.load());
         success = data->load(database, chaos_database, contributors, raptor_cache_size);
+
+        // If disruptions corruption is detected, reload data without this
+        if ((!success) && (data->disruptions_corruption_detected)) {
```

Comment:
Looks like responsibilities are mixed between data_manager and Data, only one of them should handle the errors